File: 0102-binary-tree-level-order-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {

        List<List<Integer>> res = new ArrayList<>();
        List<Integer> list = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();

        if(root == null) return res ;
        q.offer(root);
        q.offer(null);

        while(!q.isEmpty()){
            TreeNode temp = q.poll();
            if(temp == null){
                res.add(new ArrayList<>(list));
                list.clear();
                if(q.isEmpty()){
                    break;
                }else{
                   q.offer(null);
                }

            }else{
                list.add(temp.val);
                if(temp.left != null) q.offer(temp.left);
                if(temp.right != null) q.offer(temp.right);
            }


        }
        return res ;
    }
}

File: 0102-binary-tree-level-order-traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
         vector<vector<int>> v ;
    if(root == NULL) return v; 
    queue<TreeNode*> q ; 
    q.push(root);

    while(!q.empty()){
      std::vector<int> u;
      int n = q.size();

      for(int i = 0 ; i < n ; i++){
      TreeNode* k = q.front() ; q.pop();
      u.push_back(k->val);

      if(k->left != NULL)q.push(k->left);
      if(k->right != NULL)q.push(k->right);
      }
      v.push_back(u);
    }
    return v ;
    }
};
